sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -217,16 +217,21 @@ private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection c
                 }
 
                 if (connection.getErrorStream() == null) {
-                    processTextEventStream(httpRequest, httpRequestConsumer ->
-                        this.send(httpRequest), connection.getInputStream(), listener, LOGGER);
+                    processTextEventStream(httpRequest, this, connection.getInputStream(), listener, LOGGER);
                 }
             } catch (IOException e) {
                 throw LOGGER.logThrowableAsError(new UncheckedIOException(e));
             } finally {
                 connection.disconnect();
             }
         } else {
-            ResponseBodyMode responseBodyMode = httpRequest.getRequestOptions().getResponseBodyMode();
+            RequestOptions requestOptions = httpRequest.getRequestOptions();
+
+            if (requestOptions == RequestOptions.NONE) {
+                requestOptions = new RequestOptions();
+            }
```

Comment:
Same comment as above - I would ideally like to never see these checks in our code.